examples.md

```diff
@@ -28,6 +29,40 @@
   - [Swift, Objective-C - CocoaPods](#swift-objective-c---cocoapods)
   - [Swift - Swift Package Manager](#swift---swift-package-manager)
 
+## C++ -Boost
+
+Downloading, building and installing some boost libraries:
+
+```yaml
+- name: Cache Boost
+  id: cache-boost
+  uses: actions/cache@v1
+  with:
+    path: boost_1_72_0/installdir
+    key: ${{ runner.os }}-boost_1_72_0-fs-system-test
```

Comment:
I'm wondering if we can move the list of libraries to a variable:

```
env:
    boost_version: boost_1_27_0
    boost_libs: filesystem,system,test
```

So that when either changes, the `key` is in sync.  Otherwise, if either changes but the key isn't updated, this would break.

One note though is the key has a maximum length of 512 characters, so if the list of libraries grows to large this will fail.  We could hash the list of libraries to workaround the limit.